### PR TITLE
Add PIVX (prefix D)

### PR DESCRIPTION
**Pull request by guggero**
Added PIVX with values taken from https://github.com/PIVX-Project/PIVX/blob/master/src/chainparams.cpp#L171

### CODE REVIEW

**Comment by exploitagency** [comment]
Modify keyconv as well please and I will test and review.

**Comment by guggero** [comment]
Right, I forgot... I added the code to keyconv as well now.

**Comment by exploitagency** [comment]
Some comments:

Correct Line in vanitygen.c
2 should read D
https://github.com/guggero/vanitygen-plus/blob/2c60bd631dd27c3af38428322855b7ed1384173f/vanitygen.c#L472

Correct Line in keyconv.c
2 should read D
https://github.com/guggero/vanitygen-plus/blob/ce9d792e9b9523fb180dbe1413b75041616187a1/keyconv.c#L149

Change all hex values to decimal for consistency
30, 212

File: vanitygen.c [hunk 1]
```diff
@@ -469,6 +469,7 @@ main(int argc, char **argv)
 					"OMC : Omnicoin : o\n"
 					"PIGGY : Piggycoin : p\n"
 					"PINK : Pinkcoin : 2\n"
+					"PIVX : PIVX : 2\n"
```
Comment (exploitagency) [inline vanitygen.c]: Should read: "PIVX : PIVX : D\n"

File: keyconv.c [hunk 2]
```diff
@@ -146,6 +146,7 @@ main(int argc, char **argv)
 					"OMC : Omnicoin : o\n"
 					"PIGGY : Piggycoin : p\n"
 					"PINK : Pinkcoin : 2\n"
+					"PIVX : PIVX : 2\n"
```
Comment (exploitagency) [inline keyconv.c]: Should read: "PIVX : PIVX : D\n"

File: keyconv.c [hunk 3]
```diff
@@ -179,6 +180,14 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "PIVX")== 0) {
+				fprintf(stderr,
+					"Generating PIVX Address\n");
+					addrtype = 0x1E;
```
Comment (exploitagency) [inline keyconv.c]: addrtype = 30;

File: keyconv.c [hunk 4]
```diff
@@ -179,6 +180,14 @@ main(int argc, char **argv)
...
+					privtype = 0xd4;
```
Comment (exploitagency) [inline keyconv.c]: privtype = 212;

File: oclvanitygen.c [hunk 5]
```diff
@@ -260,6 +261,14 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "PIVX")== 0) {
+				fprintf(stderr,
+					"Generating PIVX Address\n");
+					addrtype = 0x1E;
+					privtype = 0xd4;
```
Comment (exploitagency) [inline oclvanitygen.c]: privtype = 212;

File: vanitygen.c [hunk 6]
```diff
@@ -502,6 +503,14 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "PIVX")== 0) {
+				fprintf(stderr,
+					"Generating PIVX Address\n");
+					addrtype = 0x1E;
+					privtype = 0xd4;
```
Comment (exploitagency) [inline vanitygen.c]: privtype = 212;

File: vanitygen.c [hunk 7]
```diff
@@ -502,6 +503,14 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "PIVX")== 0) {
+				fprintf(stderr,
+					"Generating PIVX Address\n");
+					addrtype = 0x1E;
```
Comment (exploitagency) [inline vanitygen.c]: addrtype = 30;

File: oclvanitygen.c [hunk 8]
```diff
@@ -260,6 +261,14 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "PIVX")== 0) {
+				fprintf(stderr,
+					"Generating PIVX Address\n");
+					addrtype = 0x1E;
```
Comment (exploitagency) [inline oclvanitygen.c]: addrtype = 30;

**Comment by guggero** [comment]
You're right, that was some sloppy work. I'm sorry!
I fixed it as requested.

**Comment by exploitagency** [comment]
It appears perfect now.

I will look at it again when I have had some sleep and merge it.

Thanks for your hard work!

**Pull request merged**